pkg/schedule/schedulers/hot_region_v2.go

```diff
@@ -192,15 +232,32 @@ func (bs *balanceSolver) calcProgressiveRankV2() {
 }
 
 func (bs *balanceSolver) getScoreByPriorities(dim int, rs *rankV2Ratios) int {
-	// minNotWorsenedRate, minBetterRate, minBalancedRate, maxBalancedRate, maxBetterRate, maxNotWorsenedRate can be determined from src and dst.
-	// * peersRate < minNotWorsenedRate                   ====> score == -2
-	// * minNotWorsenedRate <= peersRate < minBetterRate  ====> score == 0
-	// * minBetterRate <= peersRate < minBalancedRate     ====> score == 2
-	// * minBalancedRate <= peersRate <= maxBalancedRate  ====> score == 3
-	// * maxBalancedRate < peersRate <= maxBetterRate     ====> score == 1
-	// * maxBetterRate < peersRate <= maxNotWorsenedRate  ====> score == -1
-	// * peersRate > maxNotWorsenedRate                   ====> score == -2
-	// The higher the score, the better.
+	// For unbalanced state,
+	// roughly speaking, as long as the diff is reduced, it is either better or not worse.
+	// To distinguish the small regions, the one where the diff is reduced too little is defined as not worse,
+	// and the one where the diff is reversed is regarded as worse.
+	// For pre-balanced state,
+	// it is better only if it reach the balanced state,
+	// and it is not worse if it still in the pre-balanced state.
+	// and it is worse if it becomes the unbalanced state.
```

Comment:
can you add more info to explain states such as `pre-balanced`, `unbalanced` ?

Reply:
It has been commented in `balanceChecker`